packages/checkbox/src/useCheckbox.ts

```diff
@@ -1,23 +1,183 @@
+    },
+  })
+
+  return {
+    model,
+    isLimitExceeded,
+  }
+}
+
+const useCheckboxStatus = (props: ICheckboxProps, { model }: PartialReturnType<typeof useModel>) => {
+  const { isGroup, checkboxGroup, elFormItemSize, ELEMENT } = useCheckboxGroup()
+  const focus = ref(false)
+  const size = computed<string|undefined>(() => checkboxGroup.checkboxGroupSize?.value || elFormItemSize.value || (ELEMENT || {}).size)
+  const isChecked = computed(() => {
+    const value = model.value
+    if (toTypeString(value) === '[object Boolean]') {
+      return Boolean(value)
+    } else if (Array.isArray(value)) {
+      return value.includes(props.label)
+    } else if (value !== null && value !== undefined) {
+      return value === props.trueLabel
+    }
+  })
+  const checkboxSize = computed(() => {
+    const temCheckboxSize = props.size || elFormItemSize.value || (ELEMENT || {} as any).size
+    return isGroup.value
+      ? checkboxGroup.checkboxGroupSize?.value || temCheckboxSize
+      : temCheckboxSize
+  })
+
+  return {
+    isChecked,
+    focus,
+    size,
+    checkboxSize,
+  }
+}
+
+const useDisabled = (
+  props: ICheckboxProps,
+  { model, isChecked }: PartialReturnType<typeof useModel> & PartialReturnType<typeof useCheckboxStatus>,
+) => {
+  const { elForm, isGroup, checkboxGroup } = useCheckboxGroup()
+  const isLimitDisabled = computed(() => {
+    const max = checkboxGroup.max?.value
+    const min = checkboxGroup.min?.value
+    return !!(max || min) && (model.value.length >= max && !isChecked.value) ||
+      (model.value.length <= min && isChecked.value)
+  })
+  const isDisabled = computed(() => {
+    return isGroup.value
+      ? checkboxGroup.disabled?.value || props.disabled ||  (elForm as any || {} as any).disabled?.value || isLimitDisabled.value
+      : props.disabled || (elForm as any || {} as any).disabled?.value
+  })
+
+  return {
+    isDisabled,
+    isLimitDisabled,
+  }
+}
+
+const setStoreValue = (props: ICheckboxProps, { model }: PartialReturnType<typeof useModel>) => {
+  function addToStore() {
+    if (
+      Array.isArray(model.value) &&
+      !model.value.includes(props.label)
+    ) {
+      model.value.push(props.label)
+    } else {
+      model.value = props.trueLabel || true
+    }
+  }
+  props.checked && addToStore()
+}
+
+const useEvent = (props: ICheckboxProps, { isLimitExceeded }: PartialReturnType<typeof useModel>) => {
+  const { elFormItem } = useCheckboxGroup()
+  const { emit } = getCurrentInstance()
+  function handleChange(e: InputEvent) {
+    if (isLimitExceeded.value) return
+    const target = e.target as HTMLInputElement
+    const value = target.checked
+      ? props.trueLabel ?? true
+      : props.falseLabel ?? false
+
+    emit('change', value, e)
+  }
+
+  watch(() => props.modelValue, val => {
+    elFormItem.changeEvent?.(val)
+  })
+
+  return {
+    handleChange,
+  }
+}
+
+export const useSetAria = (props: ICheckboxProps) => {
+  const instance = getCurrentInstance()
```

Comment:
why not put the line of code into `onMounted` callback body instead of holding the reference closure when this hook gets invoked?

Reply:
I agree this point.